pubsublite/pscompat/settings.go

```diff
@@ -107,15 +107,19 @@ type PublishSettings struct {
 	// The polling interval to watch for topic partition count updates.
 	// Currently internal only and overridden in tests.
 	configPollPeriod time.Duration
+
+	// Whether compression is disabled. Compression is enabled by default.
+	DisableCompression bool
```

Comment:
Can you change this to `EnableCompression optional.Bool` for consistency with EnableIdempotence in https://github.com/googleapis/google-cloud-go/pull/7390/files (not merged yet).

Also, if you haven't already, could you ensure the integration tests pass by running locally in the `pscompat` dir: `go test -race -v`. The presubmit checks run tests with `-short`, which skips the integration tests.

Also, looks like the unit tests need updating: https://source.cloud.google.com/results/invocations/6d293971-fecf-4018-bfc8-9d753e586309/targets/cloud-devrel%2Fclient-libraries%2Fgo%2Fgoogle-cloud-go%2Fpresubmit%2Flatest-version/log

Reply:
I just merged the idempotence PR.